File: java/src/edu/berkeley/cs/util/Stack.java
package edu.berkeley.cs.util;

import java.util.Iterator;

public class Stack<T> implements Iterable<T> {
  private LinkedList<T> list;

  public Stack() {
    list = new LinkedList<>();
  }

  public Stack(LinkedList<T> list) {
    this.list = list;
  }

  public T peek() {
    return list.front();
  }

  public void push(T value) {
    list.insertFront(value);
  }

  public T pop() {
    return list.removeFront();
  }

  public int size() {
    return list.size();
  }

  public boolean isEmpty() {
    return list.isEmpty();
  }

  @Override
  public Iterator<T> iterator() {
    return list.iterator();
  }
}

File: python/src/util/stack.py
from python.src.util.linkedlist import LinkedList


class Stack:

    def __init__(self):
        self.list = LinkedList()

    def __iter__(self):
        return iter(self.list)

    def __next__(self):
        return self.list.__next__()

    def __init(self, value):
        self.list = value

    def peek(self):
        return self.list.front()

    def push(self, value):
        self.list.insert_front(value)

    def pop(self):
        return self.list.remove_front()

    def size(self):
        return self.list.size()

    def is_empty(self):
        return self.list.is_empty()
